packages/examples/react-client/src/app.tsx

```diff
@@ -121,14 +123,21 @@ export const ReactMonacoEditor: React.FC<EditorProps> = ({
             // install Monaco language client services
             MonacoServices.install();
 
-            createWebSocket(url);
+            LSPWebSocket = createWebSocket(url);
 
             return () => {
                 editorRef.current!.dispose();
             };
         }
 
-        return () => {};
+        window.onbeforeunload = () => {
+          // On page reload/exit, close web socket connection
+          !!LSPWebSocket && LSPWebSocket.close();
```

Comment:
Can we just use a modern regular syntax instead of this obfuscated code?

```typescript
if (LSPWebSocket != null) {
   LSPWebSocket.close();
}
```
Also the variable probably shouldn't start with a capital letter

Reply:
Oh, yeah, good point. The variable should be `lspWebSocket`. I don't mind the syntax, but yes, simpler is better.

Reply:
Thanks, I've renamed it to `lspWebSocket`

Reply:
@CGNonofr I did `!!lspWebSocket && lspWebSocket.close();` so that it only runs if `lspWebSocket` is defined.
Your approach works too. I just did it this way because, for my use cases, `lspWebSocket` will not always be defined, such as when the languageclient is disabled